pallets/inflation-manager/src/lib.rs

```diff
@@ -235,6 +251,54 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::set_delayed_tge())]
+		pub fn set_delayed_tge(
+			origin: OriginFor<T>,
+			block: T::BlockNumber,
+			issuance: Balance,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+
+			// Not allow to set if delayed TGE didn't enable
+			ensure!(
+				T::BlockNumber::from(0u32) != T::DoInitializeAt::get(),
+				Error::<T>::WrongDelayedTGESetting
+			);
+			ensure!(
+				InitialBlock::<T>::get() > frame_system::Pallet::<T>::block_number(),
+				Error::<T>::DelayedTGEAlreadySet
+			);
+			ensure!(
+				block > frame_system::Pallet::<T>::block_number(),
+				Error::<T>::WrongBlockSetting
+			);
+			ensure!(issuance > T::Currency::total_issuance(), Error::<T>::WrongDelayedTGESetting);
+
+			InitialBlock::<T>::put(block);
+			DoRecalculationAt::<T>::put(block);
+			TotalIssuanceNum::<T>::put(issuance);
+
+			Ok(().into())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::set_recalculation_time())]
+		pub fn set_recalculation_time(
```

Comment:
One issue I can think about is that; If we set the DoRecalculationAt flag to some desired block, lets say to end a year after only 6 months, we will face precision issues in our inflated total supply going forward. Since our block rewards were calculated for 1 whole year(in blocks).

Reply:
Other issue could be when using this extrinsic where delayed TGE is active. If DoRecalculationAt never equals InitialBlock, we could set off Inflation with default InflationParameter and incorrect TotalIssuanceNum(since TGE was not triggered).

In my opinion, we should not allow this extrinsic if delayed TGE was active.

Reply:
> One issue I can think about is that; If we set the DoRecalculationAt flag to some desired block, lets say to end a year after only 6 months, we will face precision issues in our inflated total supply going forward. Since our block rewards were calculated for 1 year(in blocks).

Yep, you are right. If sudo users use this extrinsic, they must be very sure about the inflation rate/disinflation rate and burn some tokens to ensure the total issuance number is what they want.

> In my opinion, we should not allow this extrinsic if delayed TGE was active.
That's a good point. I'll add one condition: only allow people to change the recalculation time after TGE because this extrinsic adjusts the trigger time after TGE/delay TGE.